src/lib/utils/settings/abstract_setting.hpp

```diff
@@ -37,6 +37,8 @@ class AbstractSetting : public Noncopyable, public std::enable_shared_from_this<
   virtual void unregister_at_settings_manager();
 
   const std::string name;
+ protected:
+  std::string _value;
```

Comment:
To me, the entire idea of this settings class was that settings are *not* stored as a "setting" somewhere but are dynamically retrieved whenever they are needed. Why would we need / want to store the value here?

Reply:
Apparently, I had a totally different idea then. How would that look like for an example? I added a memory budget setting to a compression example plugin: https://github.com/hyrise/hyrise/blob/2ab62566624c6cce15a773c92cdc0bc9becd9d60/src/plugins/compression_plugin.hpp

Reply:
My understanding was that the setting's `set` method would convert it to an int and change a member of the plugin itself. Similarly, a setting `SchedulerEnabledSetting` would enable the scheduler if a truthy value is passed and disable it for a falsy value. This would be hard to accomplish if everything the setting did was to store it locally.

Reply:
Okay, but then the Plugins would need public members or friend all settings, right?

Reply:
If by "all" you mean "their", then yes.

Reply:
Ok.